includes/classes/payment.php

```diff
@@ -92,30 +92,28 @@ function __construct($module = '') {
         if (!isset($credit_covers) || $credit_covers == FALSE) $_SESSION['payment'] = $include_modules[0]['class'];
       }
 
-      if ( (zen_not_null($module)) && (in_array($module, $this->modules)) && (isset($GLOBALS[$module]->form_action_url)) ) {
+      if (zen_not_null($module) && in_array($module, $this->modules) && isset($GLOBALS[$module]->form_action_url)) {
         $this->form_action_url = $GLOBALS[$module]->form_action_url;
       }
     }
   }
 
-  // class methods
-  /* The following method is needed in the checkout_confirmation.php page
+  /**
+  The update_status() method is needed in the checkout_confirmation.php page
   due to a chicken and egg problem with the payment class and order class.
-  The payment modules needs the order destination data for the dynamic status
+  The payment modules need the order destination data for the dynamic status
   feature, and the order class needs the payment module title.
-  The following method is a work-around to implementing the method in all
-  payment modules available which would break the modules in the contributions
-  section. This should be looked into again post 2.2.
+  Modules should implement this method to inspect the current order address
+  for to determine if the module should remain enabled,
+  and set their own $this->enabled property accordingly.
   */
   function update_status() {
-    if (empty($this->selected_module)) return; 
-    if (is_array($this->modules)) {
-      if (is_object($GLOBALS[$this->selected_module])) {
-        if (method_exists($GLOBALS[$this->selected_module], 'update_status')) {
-          $GLOBALS[$this->selected_module]->update_status();
-        }
-      }
-    }
+    if (empty($this->selected_module)) return;

```

Comment:
Any reason for the multi-if-statements instead of aggregating on a single `if`?

Reply:
Largely easier readability.

Reply:
To clarify: the "early return" concept allows one to quickly see all the criteria that do not apply to the outcome, and then see only the actual real "actions" once all those disqualifiers are out of the way.
Several short `if` statements that just directly return/abort makes human reading of it very quick.

Reply:
Different strokes for different folks.